### PR TITLE
feat: support adding custom konva nodes

### PR DESCRIPTION
**Pull request by ComfyFluffy**
Supporting using custom nodes by pass a `customNodes` property into options.

A few changes are made:
- Run Prettier format for the whole project.
- Get name directly from node constructors. 
- Add an option that takes a `new (...args: any) => Node<any>` array.
- Use `Node<any>` instead of union types.

Related issue: #200

### CODE REVIEW

**Comment by ComfyFluffy** [comment]
The changes are yet tested. We will test them soon on on our project.

File: src/index.ts [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
+      Konva.Wedge,
+      ...(options?.customNodes || []),
+    ];
 
-    [Stage, ...KONVA_NODES.map(KonvaNode)].map((k) => {
-      app.component(`${prefixToUse}${k.name}`, k);
+    const components: Component[] = [
+      Stage,
+      ...konvaNodeConstructors.map((konvaNodeConstructor) =>
+        KonvaNode(konvaNodeConstructor.name, konvaNodeConstructor),
```
Comment (nighca) [inline src/index.ts:45]: `KonvaNodeConstructor` may be renamed within code-building (mangling), which makes its `name` unreliable.

Comment (ComfyFluffy) [reply src/index.ts:45]: Indeed. I will make changes so it takes a Record<string, KonvaNodeConstructor>

**Comment by nighca** [comment]
@ComfyFluffy I think it's better to link the related issues (such as #200) in your PR description.

**Comment by ComfyFluffy** [comment]
We have tested the changes in our project.

**Comment by nighca** [comment]
@lavrton Plz have a look on this PR.

**Comment by lavrton** [comment]
Released as 3.1.0. Thanks for the good PR. Let me know how it works.

**Pull request merged**